ncm-shorewall/src/main/pan/components/shorewall/schema.pan

```diff
@@ -22,7 +22,21 @@ type component_shorewall_masq = {
     "origdest" ? string
     "probability" ? double(0..1)
 };
-
+# Keep this list in sync with list from TT file
+@{a snat entry: ACTION SOURCE DEST PROTO PORT IPSEC MARK USER SWITCH ORIGDEST PROBABILITY}
+type component_shorewall_snat = {
+    "action" : string with match(SELF, '^(CONTINUE|LOG|MASQUERADE|NFLOG|SNAT|ULOG)') # can have trailing parameters
+    "source" ? string
+    "dest" ? string[]
+    "proto" ? string
+    "port" ? string[]
+    "ipsec" ? string[]
+    "mark" ? string
+    "user" ? string
+    "switch" ? string
+    "origdest" ? string
```

Comment:
Could any of these `string` types be restricted further?
Perhaps just by using `string_trimmed` to prevent leading or trailing white-space?